.circleci/config.yml

```diff
@@ -1,72 +1,327 @@
+        type: string
+    steps:
+      - unless:
+          condition: <<parameters.pr>>
+          steps:
+            - run:
+                name: Publish image for <<parameters.registry>>
+                command: >
+                  IFS="," read -ra DOCKER_REGISTRIES \<<< "<< parameters.registry >>"
 
-      - run:
-          name: Launching container for testing
-          command: |
-            docker network create testnet
-            docker run --name bind-server -d --net testnet $IMAGE_NAME
-            sleep 5
+                  for registry in "${DOCKER_REGISTRIES[@]}"; do
+                    IFS="," read -ra DOCKER_TAGS \<<< "<< parameters.tag >>"
+
+                    for tag in "${DOCKER_TAGS[@]}"; do
+                      docker push $registry/<< parameters.image>>:${tag}
+                    done
+                  done
+
+jobs:
+  build:
+    executor: docker/machine
+    steps:
+      - checkout
+      - docker-build:
+          registry: docker.io,quay.io
+          image: sameersbn/bind
+          tag: ${CIRCLE_TAG:-latest}
+          cache_from: docker.io/sameersbn/bind:latest
+      - docker-save:
+          registry: docker.io,quay.io
+          image: sameersbn/bind
+          tag: ${CIRCLE_TAG:-latest}
 
+  test:
+    executor: docker/machine
+    steps:
+      - checkout
+      - docker-load
+      - run:
+          name: Create test network
+          command: docker network create testnet
+      - run:
+          name: Launch bind container
+          command: docker run --name bind -d --net testnet sameersbn/bind:${CIRCLE_TAG:-latest}
+      - run:
+          name: Wait for container bootup
+          command: sleep 15
+      - run:
+          name: Container info
+          command: docker ps -a
       - run:
-          name: Testing image
+          name: Test image
           command: |
-            docker run --rm --net testnet $IMAGE_NAME host www.google.com bind-server
+            docker run --rm --net testnet sameersbn/bind:${CIRCLE_TAG:-latest} host www.google.com bind
 
+  publish-dockerhub:
+    executor: docker/machine
+    steps:
+      - docker-load
+      - docker/check:
+          registry: docker.io
+          docker-username: DOCKER_LOGIN
+          docker-password: DOCKER_PASSWORD
+      - docker-publish:
+          registry: docker.io
+          image: sameersbn/bind
+          tag: ${CIRCLE_TAG:-latest}
+
+  publish-quay:
+    executor: docker/machine
+    steps:
+      - docker-load
+      - docker/check:
+          registry: quay.io
+          docker-username: DOCKER_LOGIN
+          docker-password: DOCKER_PASSWORD
+      - docker-publish:
+          registry: quay.io
+          image: sameersbn/bind
+          tag: ${CIRCLE_TAG:-latest}
+
+  release:
+    executor:
+      name: go/default
+      tag: '1.14'
+    steps:
+      - checkout
+      - run:
+          name: Installing github-release tool
+          command: go get github.com/meterup/github-release
```

Comment:
Do we really need to use `meterup/github-release` fork instead of https://github.com/github-release/github-release ?

It seems there are no difference between upstream and fork repos.